ms/MSOper/MSKeys.cc

```diff
@@ -161,6 +161,24 @@ std::set<ArrayKey> uniqueArrayKeys(const std::set<ScanKey>& scanKeys) {
     return arrayKeys;
 }
 
+std::set<ScanKey> filter(const std::set<ScanKey> scans, const ArrayKey& arrayKey) {
+    std::set<ScanKey> subset;
+    std::set<ScanKey>::const_iterator iter = scans.begin();
+    std::set<ScanKey>::const_iterator end = scans.end();
+    ArrayKey arrayFromScan;
+    for (; iter!=end; ++iter) {
```

Comment:
the function can take different typed sets, but you need a ScanKey vs ArrayKey comparison function
if that makes no sense outside of this function its probably not worthwhile to add, then this is fine